pkgs/top-level/all-packages.nix

```diff
@@ -13318,15 +13318,7 @@ in
 
   judy = callPackage ../development/libraries/judy { };
 
-  kdeFrameworks =
-    let
-      mkFrameworks = import ../development/libraries/kde-frameworks;
-      attrs = {
-        inherit libsForQt5;
-        inherit lib fetchurl;
-      };
-    in
-      recurseIntoAttrs (makeOverridable mkFrameworks attrs);
+  inherit (libsForQt5) kdeFrameworks;
```

Comment:
Exposing a default `kdeFrameworks` attribute at the top level is going to perpetuate the problem we already have with packages importing dependencies from the wrong set. All Qt packages that depend on KDE Frameworks must import those libraries with `libsForQt5*`, so why should this be here?

Reply:
Right. I added this here for backwards-compatibility.

I think we should get rid of it, and link to each of the applications here, via `qt-packages.nix`. So, you would something like `inherit (libsForQt5) dolphin;`, which is like how it was before.

It means undoing use of `aliases.nix` for the applications, which I think in retrospect was an incorrect solution.

Reply:
I agree with @ttuegel on this one as well.

> It means undoing use of `aliases.nix` for the applications, which I think in retrospect was an incorrect solution.

I did it in order to avoid future usage of attributes to be used as inputs from the wider set, vs as inputs from the correct `libsForQt5*` set. It doesn't have to be that way if they'd all be part of qt5-packages.nix.

Reply:
I have removed the top-level `kdeFrameworks`, `kdeApplications` and `plasma5`, however, I did add in return a `plasma5Packages` which maps to the version of `libsForQt5` that is used for `plasma5`, mainly for NixOS.